backend/tournesol/suggestions/suggester.py

```diff
@@ -0,0 +1,236 @@
+        self._entity_to_video = {}
+        self._user_specific_graphs = {}
+        self.poll = concerned_poll
+        self.criteria = self.poll.criterias_list[0]
+        # build complete graph
+        comparison_queryset: QuerySet = ComparisonCriteriaScore.objects \
+            .filter(comparison__poll__name=self.poll.name) \
+            .filter(criteria=self.criteria)
+        self._complete_graph = Graph(None, self.poll, self.criteria)
+
+        for c in comparison_queryset:
+            # Checks if each compared Entity has already been translated to a SuggestedVideo object
+            # and translates it otherwise
+            if c.comparison.entity_1 not in self._entity_to_video.keys():
+                self._entity_to_video[c.comparison.entity_1] = SuggestedVideo(
+                    self._comparison_reference_video, from_entity=c.comparison.entity_1
+                )
+
+                self._complete_graph.add_node(
+                    self._entity_to_video[c.comparison.entity_1]
+                )
+
+            if c.comparison.entity_2 not in self._entity_to_video.keys():
+                self._entity_to_video[c.comparison.entity_2] = SuggestedVideo(
+                    self._comparison_reference_video, from_entity=c.comparison.entity_2
+                )
+
+                self._complete_graph.add_node(
+                    self._entity_to_video[c.comparison.entity_2]
+                )
+
+            self._complete_graph.add_edge(
+                self._entity_to_video[c.comparison.entity_1],
+                self._entity_to_video[c.comparison.entity_2],
+            )
+
+        # create required user graphs (none at first in fact)
+
+    def _get_user_comparability_augmenting_videos(self) -> list[SuggestedVideo]:
+        """"
+        Function called to get the videos to recommend while the scale and translation
+        uncertainties of the user are not high enough
+        """
+        # I want all entities from the current poll compared by supertrusted user
+        # todo create alias to properly detect supertrusted ?
+        req_entities = Entity.objects \
+            .filter(Q(comparisons_entity_1__poll__name=self.poll.name) |
+                    Q(comparisons_entity_2__poll__name=self.poll.name))\
+            .filter(Q(comparisons_entity_1__user__is_staff=True) |
+                    Q(comparisons_entity_2__user__is_staff=True))\
+            .distinct()
+
+        return list(map(lambda entity: self._entity_to_video[entity], req_entities))
+
+    def _get_user_rate_later_video_list(self, user: User) -> list[SuggestedVideo]:
+        """
+        Function to get the list of videos of the user's rate later list
+        """
+        req_entities = Entity.objects \
+            .filter(type=self.poll.name) \
+            .filter(videoratelater__user__email=user.email)
+        return list(map(lambda entity: self._entity_to_video[entity], req_entities))
```

Comment:
Still useful?

Reply:
It is not yet used, but will, I should take it into account to recommend videos